app/controllers/solidus_square/callback_actions_controller.rb

```diff
@@ -8,12 +8,13 @@ class CallbackActionsController < BaseController
     before_action :load_order, only: [:square_checkout]
 
     def square_checkout
-      authorize! :update, @order, order_token
+      authorize! :update, @order, cookies.signed[:guest_token]
 
       checkout_page_url = helpers.solidus_square_gateway.checkout(@order, redirect_url)[:checkout_page_url]
 
       respond_to do |format|
         format.html { redirect_to checkout_page_url }
+        format.json { render json: { checkout_page_url: checkout_page_url } }
```

Comment:
We have a [specific controller](https://github.com/nebulab/solidus_square/blob/main/app/controllers/solidus_square/api/checkout_controller.rb#L5) to respond to JSON. Why you have added it back here?

Reply:
https://github.com/nebulab/solidus_square/pull/49/commits/930584fd6ac9faa791f751ac32ef34093bf43ee1